### PR TITLE
Feat/everstake modal

### LINKED ISSUE #9267: Staking: Confirm dialog + Sign TX
### User Story

As an Ethereum user
I want to stake Ethereum
So that I can earn a return on my hodlings

### Acceptance Criteria

Background: 'Add to staking pool' value is dynamic

Given I have 1 ETH to stake on 'Stake Ethereum' form
And I have sufficient balance
When I click 'Continue'
Then I should see 'Confirm'

Given I see 'Confirm'
Then I should see '35 days'
And I should see 'Confirm to stake' is disabled

Given I see 'Confirm'
When I click on 'I acknowledge the above about staking'
Then I should see 'Confirm to stake' is enabled

Given 'Confirm to stake' is enabled
When I click on 'Confirm to stake'
Then I should see 'Confirm on Trezor'

Given I see 'Confirm on Trezor'
Then I should see 'Amount to stake'
And I should see '1 ETH'

Given I am in tx review modal
When I click on 'X' (Cancel)
Then It should take me back to staking form modal

### [Screens](https://www.figma.com/file/hZ22ckyTRlCYRLTNUbTdWO/Exploring-Accounts?type=design&node-id=6218-24052&mode=design&t=biDstCGFMVCBHo23-4)
image
image

### LINKED ISSUE #11256: Staking set dressing: Fund leaving Trezor
# User Story
As an Ethereum User
I want to know the risks of staking on Trezor
So that I can evaluate any security concerns

# Acceptance Criteria
Given I am on the Staking in a nutshell modal
When I click continue
Then I should see 'Your staked funds are managed by...'

# Design
https://www.figma.com/file/0Kz3w6WDlvQm62RjOm87na/Staking?type=design&node-id=4285-21022&mode=design&t=52EcRrIDbhO337OD-4

### LINKED ISSUE #11407: Staking: Dynamic APY
## User story
As a user
I want to know the APY of staking
So that I can gauge the return of my investment

## Acceptance criteria
Given I am on the staking tab
When I have no ETH staked
Then I should see the current APY

Given I am on the staking tab
When I have ETH staked
Then I should see 'Annual Percentage Yield' 
And I should see the current APY

## Screens

apy-welcome-screen
Screenshot 2024-03-01 at 17 20 13

### PR DESCRIPTION
**Pull request by StanCheb**
## Description

1. The waiting period in the "Confirm staking period" modal is now dynamic. (NB! Do not forget to switch to the prod API before release)
2. The Everstake modal is added.

## Related Issue

Resolve 
closes https://github.com/trezor/trezor-suite/issues/9267
closes https://github.com/trezor/trezor-suite/issues/11256
closes [#11407](https://github.com/trezor/trezor-suite/issues/11407)

## Screenshots:
Знімок екрана 2024-03-19 о 12 02 15
Знімок екрана 2024-03-19 о 12 01 25

### CODE REVIEW

**Comment by sime** [comment]
> NB! Do not forget to switch to the prod API before release)
@shotgunofdeath @tomasklim https://github.com/trezor/trezor-suite/issues/11407#issuecomment-1999236771

Can we have both staging and production endpoints in the app?

**Comment by StanCheb** [comment]
Yes, we can. The staging API might be used for Goerli/Holesky and the prod one for real ETH

**Comment by sime** [comment]
@shotgunofdeath Can that be included in this PR? Or is it off-topic ?

**Comment by StanCheb** [comment]
@sime It surely can. Just not sure when exactly

File: packages/suite/src/support/messages.ts [hunk 1]
```diff
@@ -8792,4 +8792,37 @@ export default defineMessages({
...
         defaultMessage: 'Provided by',
     },
+    TR_STAKE_YOUR_FUNDS_MANAGED_1: {
+        id: 'TR_STAKE_YOUR_FUNDS_MANAGED_1',
+        defaultMessage: 'Your staked funds are',
+    },
+    TR_STAKE_YOUR_FUNDS_MANAGED_2: {
+        id: 'TR_STAKE_YOUR_FUNDS_MANAGED_2',
+        defaultMessage: 'managed by Everstake',
+    },
```
Comment (tomasklim) [inline packages/suite/src/support/messages.ts]: Please merge those separated strings into one and use similar logic to e.g. translation `TR_MOBILE_APP_PROMO_TEXT_FOOTER`

```
  <Translation
      values={{ h: text => <HeadingAccent>{text}</HeadingAccent> }}
      id="TR_STAKE_YOUR_FUNDS_MANAGED"
  />

    TR_STAKE_YOUR_FUNDS_MANAGED: {
        id: 'TR_STAKE_YOUR_FUNDS_MANAGED',
        defaultMessage: 'Your staked funds are <h>managed by Everstake<h>',
    },
```

**Review by tomasklim — CHANGES_REQUESTED** [review]

**Comment by tomasklim** [comment]
@sime does it makes sense to have staging url for testnets? I think we do not care about APY there and validator's queue, but I dont know

but sure it is easy to change it to use staging for testnets only, and prod for mainnet, i can do that right away

**Comment by tomasklim** [comment]
efb62547a051a06b034bf765e8fcd3587c01b660 @shotgunofdeath does this make sense?

**Comment by sime** [comment]
@tomasklim thinking about the staking delay etc. not just apy

File: packages/suite/src/support/messages.ts (hunk 1 above)
Comment (tomasklim) [reply packages/suite/src/support/messages.ts]: @shotgunofdeath i will fix it

**Review by tomasklim — APPROVED** [review]

**Comment by StanCheb** [comment]
> [efb6254](https://github.com/trezor/trezor-suite/commit/efb62547a051a06b034bf765e8fcd3587c01b660) @shotgunofdeath does this make sense?

Yes, totally

**Pull request merged**